<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="arm"/>
  <arg name="end_effector"/>
  <arg name="ft_sensor"/>
  <arg name="laser_model"/>
  <arg name="camera_model"/>
  <arg name="multi" default=""/>
  <arg name="model_name" default="tiago"/>
  <arg name="file_suffix" value="$(eval ('no-arm' if not arm else end_effector+ ('' if not ft_sensor else '_'+ft_sensor)))"/>
  <rosparam ns="pal_robot_info" param="type">"tiago"</rosparam>

  <!-- PID gains -->
  <rosparam command="load" file="$(find tiago_controller_configuration_gazebo)/config/pids.yaml"/>

  <rosparam command="load" file="$(find tiago_controller_configuration_gazebo)/config/hardware_gazebo/tiago_pal_hardware_gazebo_$(arg file_suffix).yaml"/>

  <!-- Joystick controller parameters -->
  <!--
  <rosparam file="$(find tiago_gazebo)/config/joy_teleop_sim.yaml" command="load"/>
  -->

  <!-- Robot model -->
  <include file="$(find tiago_description)/robots/upload.launch">
    <arg name="arm" value="$(arg arm)"/>
    <arg name="end_effector" value="$(arg end_effector)"/>
    <arg name="ft_sensor" value="$(arg ft_sensor)"/>
    <arg name="laser_model" value="$(arg laser_model)"/>
    <arg name="camera_model" value="$(arg camera_model)"/>
    <arg name="multi"  value="$(arg multi)"/>
  </include>

  <!-- Spawn robot in Gazebo -->
  <node output='screen' pkg="my_simulation" type="spawn_robot.py" name="spawn_robot"
    args="-urdf -param robot_description -model $(arg model_name)" />

</launch>
